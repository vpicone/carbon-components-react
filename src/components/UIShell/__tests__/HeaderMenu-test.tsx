/**
 * Copyright IBM Corp. 2016, 2018
 *
 * This source code is licensed under the Apache-2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { mount } from 'enzyme';
import { HeaderMenu, HeaderMenuItem } from '../';

describe('HeaderMenu', () => {
  let mountNode;
  let mockProps;

  beforeEach(() => {
    mountNode = document.createElement('div');
    mockProps = {
      'aria-label': 'Accessibility label',
      className: 'custom-class',
      // We use `ref` instead of `focusRef` becase `HeaderMenu` forwards the ref
      // to the underlying menu button
      ref: jest.fn(),
      tabIndex: -1,
    };

    document.body.appendChild(mountNode);
  });

  afterEach(() => {
    mountNode.parentNode.removeChild(mountNode);
  });

  it('should render', () => {
    const wrapper = mount(
      <HeaderMenu {...mockProps}>
        <HeaderMenuItem href="/a">A</HeaderMenuItem>
        <HeaderMenuItem href="/b">B</HeaderMenuItem>
        <HeaderMenuItem href="/c">C</HeaderMenuItem>
      </HeaderMenu>,
      {
        attachTo: mountNode,
      }
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('menu button', () => {
    it.skip('should set the given ref on the menu button', () => {});

    it.skip('should open and close the menu when interacted with by a mouse', () => {});

    it.skip('should focus the first item in the menu if DOWN is pressed', () => {});

    it.skip('should focus the last item in the menu if UP is pressed', () => {});
  });

  describe('menu', () => {
    // https://www.w3.org/TR/wai-aria-practices/examples/menubar/menubar-1/menubar-1.html#kbd2_label
    it.skip('should close the menu if RIGHT or LEFT are pressed', () => {});

    it.skip('should close the menu and set focus on the menu button if ESC is pressed', () => {});

    it.skip('should focus the first item if HOME is pressed', () => {});

    it.skip('should focus the last item if END is pressed', () => {});

    it.skip('should support movement with UP and DOWN keys', () => {});

    it.skip('should close the menu if blur occurs that is not in the menu or menubar', () => {});
  });
});
