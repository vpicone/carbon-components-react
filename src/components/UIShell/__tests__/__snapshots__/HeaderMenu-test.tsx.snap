// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu should render 1`] = `
<ForwardRef
  aria-label="Accessibility label"
  className="custom-class"
  tabIndex={-1}
>
  <HeaderMenu
    aria-label="Accessibility label"
    className="custom-class"
    focusRef={null}
    tabIndex={-1}
  >
    <li
      className="bx--header__submenu custom-class"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
    >
      <a
        aria-expanded={false}
        aria-haspopup="menu"
        className="bx--header__menu-item bx--header__menu-title"
        href="javascript:void(0)"
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex={0}
      >
        Accessibility label
        <ChevronDownGlyph
          className="bx--header__menu-arrow"
          height={6}
          preserveAspectRatio="xMidYMid meet"
          viewBox="0 0 10 6"
          width={10}
          xmlns="http://www.w3.org/2000/svg"
        >
          <svg
            aria-hidden={true}
            className="bx--header__menu-arrow"
            focusable="false"
            height={6}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 10 6"
            width={10}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 6L0 1 .7.3 5 4.6 9.3.3l.7.7z"
            />
          </svg>
        </ChevronDownGlyph>
      </a>
      <ul
        aria-label="Accessibility label"
        className="bx--header__menu"
        role="menu"
      >
        <ForwardRef
          href="/a"
          key=".0"
          role="none"
        >
          <HeaderMenuItem
            href="/a"
            innerRef={[Function]}
            role="none"
          >
            <li
              role="none"
            >
              <Link
                className="bx--header__menu-item"
                element="a"
                href="/a"
                role="menuitem"
                tabIndex={0}
              >
                <a
                  className="bx--header__menu-item"
                  href="/a"
                  role="menuitem"
                  tabIndex={0}
                >
                  <span
                    className="bx--text-truncate--end"
                  >
                    A
                  </span>
                </a>
              </Link>
            </li>
          </HeaderMenuItem>
        </ForwardRef>
        <ForwardRef
          href="/b"
          key=".1"
          role="none"
        >
          <HeaderMenuItem
            href="/b"
            innerRef={[Function]}
            role="none"
          >
            <li
              role="none"
            >
              <Link
                className="bx--header__menu-item"
                element="a"
                href="/b"
                role="menuitem"
                tabIndex={0}
              >
                <a
                  className="bx--header__menu-item"
                  href="/b"
                  role="menuitem"
                  tabIndex={0}
                >
                  <span
                    className="bx--text-truncate--end"
                  >
                    B
                  </span>
                </a>
              </Link>
            </li>
          </HeaderMenuItem>
        </ForwardRef>
        <ForwardRef
          href="/c"
          key=".2"
          role="none"
        >
          <HeaderMenuItem
            href="/c"
            innerRef={[Function]}
            role="none"
          >
            <li
              role="none"
            >
              <Link
                className="bx--header__menu-item"
                element="a"
                href="/c"
                role="menuitem"
                tabIndex={0}
              >
                <a
                  className="bx--header__menu-item"
                  href="/c"
                  role="menuitem"
                  tabIndex={0}
                >
                  <span
                    className="bx--text-truncate--end"
                  >
                    C
                  </span>
                </a>
              </Link>
            </li>
          </HeaderMenuItem>
        </ForwardRef>
      </ul>
    </li>
  </HeaderMenu>
</ForwardRef>
`;
